src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultOfStates = [];
+  const newState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      Object.assign(newState, actions[i].extraData);
+      resultOfStates.push({ ...newState });
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      for (let y = 0; y < actions[i].keysToRemove.length; y++) {
+        delete newState[actions[i].keysToRemove[y]];
+      }
+      resultOfStates.push({ ...newState });
+    }
+
+    if (actions[i].type === 'clear') {
+      for (const key in newState) {
+        delete newState[key];
+      }
+      resultOfStates.push({ ...newState });
+    }
```

Comment:
This part is too complex, it would be better to maintain it with the switch/case